src/main/java/org/graylog/plugins/usagestatistics/UsageStatsOptOutService.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    protected UsageStatsOptOutService(ClusterConfigService clusterConfigService,
+                                      UsageStatsConfiguration config,
+                                      OkHttpClient httpClient,
+                                      ObjectMapper objectMapper,
+                                      Executor executor) {
+        this.clusterConfigService = clusterConfigService;
+        this.config = config;
+        this.executor = executor;
+        this.httpClient = httpClient;
+        this.objectMapper = objectMapper;
+    }
+
+    public UsageStatsOptOutState getOptOutState() {
+        final UsageStatsOptOutState optOutState = clusterConfigService.get(UsageStatsOptOutState.class);
+
+        return optOutState == null ? UsageStatsOptOutState.create(false) : optOutState;
+    }
+
+    public void createOptOut() {
+        // Run the opt-out request outside of the calling thread so it does not block.
+        executor.execute(new Runnable() {
+            @Override
+            public void run() {
+                clusterConfigService.write(UsageStatsOptOutState.create(true));
+
+                final URL url = getUrl();
+
+                if (url == null) {
+                    LOG.debug("Not sending opt-out request, 'usage_statistics_url' is not set.");
+                    return;
+                }
+
+                final Headers headers = new Headers.Builder()
+                        .add(HttpHeaders.USER_AGENT, USER_AGENT)
+                        .add("X-Usage-Statistics-Version", USAGE_STATS_VERSION)
+                        .build();
+
+                final Request request = new Request.Builder()
+                        .url(url)
+                        .headers(headers)
+                        .post(RequestBody.create(CONTENT_TYPE, buildPayload()))
+                        .build();
+
+                final Response response;
+                try {
+                    response = httpClient.newCall(request).execute();
+                } catch (IOException e) {
+                    LOG.error("Error while sending anonymous usage statistics opt-out");
+                    LOG.debug("Error details", e);
+                    return;
+                }
+
+                if (!response.isSuccessful()) {
+                    LOG.warn("Couldn't successfully send usage statistics opt-out: {}", response);
+                }
+            }
+        });
+    }
+
+    @Nullable
+    protected URL getUrl() {
+        final ClusterId clusterId = clusterConfigService.get(ClusterId.class);
+
+        if (clusterId != null) {
+            return HttpUrl.get(config.getUrl()).newBuilder()
+                    .addPathSegment("cluster")
+                    .addPathSegment(clusterId.clusterId())
+                    .addPathSegment("optout")
+                    .build()
+                    .url();
+        } else {
+            return null;
+        }
+    }
+
+    protected byte[] buildPayload() {
```

Comment:
Minor nitpick: Adding `@Nullable` might help the IDE to do some code inspections.